### PR TITLE
Create Upload-file

### PR DESCRIPTION
**Pull request by sgwoodford**
I dont know java script, but used Chat GPT to crate this code, using the existing Download-file code as a template.

## WHAT

Upload files to onedrive

### <samp>🤖 Generated by Copilot at 8fe363a</samp>

* Add a new action to upload a local file to OneDrive ([link](https://github.com/PipedreamHQ/pipedream/pull/7146/files?diff=unified&w=0#diff-e0730135707845d6ebf53b0020e853a07a33f37f15cd363c48865ffc9a6878ddR1-R58))

### CODE REVIEW

**Comment by pipedream-component-development** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by pipedream-component-development** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/microsoft_onedrive/actions/Upload-file [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  version: "0.0.1",
+  type: "action",
+  props: {
+    onedrive,
+    uploadFolderId: {
+      type: "string",
+      label: "Upload Folder ID",
+      description: "The ID of the folder where you want to upload the file.",
+      optional: false,
+    },
```
Comment (luancazarine) [inline components/microsoft_onedrive/actions/Upload-file:19]: Here you must to add the async options to list the folders. You can see an example [here](https://github.com/PipedreamHQ/pipedream/blob/33d11f0cb100c7c29a30f45f860abce7dca95481/components/github/github.app.mjs#L55)

**Review by luancazarine — CHANGES_REQUESTED** [review]
Hi @sgwoodford, thanks for your contribution. I just left a comment.

**Comment by michelle0927** [comment]
Resolved by https://github.com/PipedreamHQ/pipedream/pull/7905

**Pull request closed without merging**